parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java

```diff
@@ -279,6 +279,11 @@ public ParquetWriter(Path file, Configuration conf, WriteSupport<T> writeSupport
     WriteSupport.WriteContext writeContext = writeSupport.init(conf);
     MessageType schema = writeContext.getSchema();
 
+    // encryptionProperties could be built from the implementation of EncryptionPropertiesFactory when it is attached.
+    if (encryptionProperties == null) {
+      encryptionProperties = ParquetOutputFormat.createEncryptionProperties(conf, new Path(file.toString()), writeContext);
```

Comment:
This assumes that `file.toString()` returns the full file path. However, the `file` is a `public abstract interface org.apache.parquet.io.OutputFile`, which doesn't have such method, so `toString()` is up to the implementation; no guarantees it will return the path. Also, `new Path(string full_path)` is not aware of the right filesystem (?) Maybe can be handled with an upcast to a known implementing class - preferably one that already has a `Path getPath()` method. 
But of course, this won't be very general.

Reply:
eg upcasting to `HadoopOutputFile`. Or even better, adding `Path getPath()` method to the `OutputFile` - this should be general enough.

Reply:
Good point. I like the idea of adding "Path getPath()" better. Since this feature is going to be in a major release. I think
adding a new method to the interface could be fine.

Reply:
looks good. please also replace `new Path(file.toString())` with `file.getPath()`